rpc/src/v1/impls/light/eth.rs

```diff
@@ -420,15 +420,24 @@ where
 	}
 
 	fn transaction_by_hash(&self, hash: H256) -> BoxFuture<Option<Transaction>> {
-		{
-			let tx_queue = self.transaction_queue.read();
-			if let Some(tx) = tx_queue.get(&hash) {
+		let in_txqueue = self.transaction_queue.read().get(&hash).is_some();
+
+		// The transaction is in the `local txqueue`
+		// Check if the nonce on the accounts are still fresh and return result from
+		// the `local txqueue` if it is still fresh.
```

Comment:
If `light_cull` call fails, we still return result from the `local txqueue`.  Should we change or document that?

Reply:
I think it should return the local transaction if that's the case I tried to change to docs let me know if it is clear enough

Reply:
I think the code is quite self-explanatory and the `If it fails return the local transaction` is slightly misleading (may read as we return the local transaction only in this case), so I suggest to remove this part.